Enhancements and related code cleanup:

jeffriley opened the pull request:
(1) implemented DETAILED_OUTPUT folder inside COMPAS_Output container for SSE output
(2) SSE Parameters files moved to DETAILED_OUTPUT folder (they are analogous to BSE_Detailed_Output files)
(3) implemented SSE Switch Log and BSE Switch Log files (record written at the time of stellar type switch - see documentation)
(4) implemented SSE Supernova log file - see documentation (issue #253)
(5) added TIMESCALE_MS as a valid property in BaseStar::StellarPropertyValue().  The TIMESCALE_MS value in the SSE_Parameters file was being printed as ERROR! and nobody noticed :-)  It now prints correctly.

Rudimentary testing done - files created correctly.  No changes made to anything other than logging code (except for trivial change for (5) above), so contents of logfiles should be unchanged. More rigorous testing is always a good thing, so reviewers please test rigorously :-)

@ilyamandel Switch Log files here - feel free to change the default record specifications.  Like other log files they can be changed at run time, so the defaults are not so important.  Use new program options --BSEswitchLog and --SSEswitchLog to enable.

@reinhold-willcox @SimonStevenson SSE Supernova file is here - feel free to change the default record specifications.  Like other log files they can be changed at run time, so the defaults are not so important.

jeffriley committed:
Enhancements and related code cleanup:
(1) implemented DETAILED_OUTPUT folder inside COMPAS_Output container for SSE output
(2) SSE Parameters files moved to DETAILED_OUTPUT folder (they are analogous to BSE_Detailed_Output files)
(3) implemented SSE Switch Log and BSE Switch Log files (record written at the time of stellar type switch - see documentation)
(4) implemented SSE Supernova log file - see documentation (issue #253)
(5) added TIMESCALE_MS as a valid property in BaseStar::StellarPropertyValue().  The TIMESCALE_MS value in the SSE_Parameters file was being printed as ERROR! and nobody noticed :-)  It now prints correctly.

jeffriley committed:
Adding documentation pdf

SimonStevenson commented on src/GiantBranch.cpp:
```diff
@@ -1721,6 +1721,15 @@ STELLAR_TYPE GiantBranch::ResolveSupernova() {
+
+        // stash SN details for later printing to the SSE Supernova log
+        // can't print it now because we may revert state (in Star::EvolveOneTimestep())
+        // will be printed in tar::EvolveOneTimestep() after timestep is accepted (i.e. we don't revert state)
```
Typo : Star
```suggestion
        // will be printed in Star::EvolveOneTimestep() after timestep is accepted (i.e. we don't revert state)
```

SimonStevenson commented on src/Log.cpp:
```diff
@@ -986,6 +986,181 @@ std::tuple<bool, LOGFILE> Log::GetStandardLogfileKey(const int p_FileId) {
+ * write, we might have a file created that may never have data records written to it (we
+ * could specifically check for that and delete the file, but that's a little inelegant -
+ * better to not create the file in the first place).  So, this function will enable me to 
+ * format a SSE Supernova record at the righ time, but delay writing it to after we decide 
```
Typo right
```suggestion
 * format a SSE Supernova record at the right time, but delay writing it to after we decide 
```

SimonStevenson commented on docs/COMPAS_LaTex/sections/UserGuide/COMPASOutput.tex:
```diff
@@ -0,0 +1,239 @@
+\begin{itemize}
+\itemsep0pt
+\item{SSE Parameters log file \\ Records detailed information for a star during SSE.}
+\item{SSE Switch log file \\ Records detailed information for a star at the time of each stellar type switch during SSE. Enable with program option \textit{\texttt{-{}-{BSEswitchLog}}}.}
```
I think this should be SSE
```suggestion
\item{SSE Switch log file \\ Records detailed information for a star at the time of each stellar type switch during SSE. Enable with program option \textit{\texttt{-{}-{SSEswitchLog}}}.}
```

SimonStevenson reviewed (changes requested):
Looks good Jeff, I compiled and ran a bunch of single stars and checked that the SSE_Supernova file was created, and seems to look good. 

Do we need to add 

```
logfile_SSE_supernova = None
logfile_SSE_detailed_output = None
```

to `pythonSubmitDefault.py`? 

I also highlighted a few typos in some of the files below.

I found it a bit difficult to read through this changed, since there are many changed files (52!)

jeffriley committed:
Typos addressed, pythonSubmitDefault.py updated

jeffriley commented:
Quite right Simon - thanks!  I've addressed the typos and updated pythonSubmitdefault.py.

Yes, 52 files changed, but most of those are the new LaTex documentation - you probably don't need to review the source of those - just check the PDF.

jeffriley commented:
Just noticed this:

> logfile_SSE_detailed_output = None

SSE detailed output is SSE_Parameters files - there is no mechanism to turn them off.  Up until now, they were the only artefact of SSE - but now the SSE Supernova file will be created, and the user can select the SEE Switch Log.

Given that we now have the SSE Supernova file, should we add an option to disable toe SSE_Parameters file - i.e. the SSE detailed output?
